### PR TITLE
Automatically generate Go interfaces code, etc.

### PR DESCRIPTION
**Pull request by canstand**
Due to the rewrite of the go code generation scripts, it brings the following **breaking** changes:

- Optimized generated comments. Links and **Deprecated** tags in Go doc comments now work.
  - May cause many **Deprecated** lint errors, please update the call or use `//nolint:staticcheck` to ignore.
- Added event interface methods. For example `Page.OnDialog()` etc.
- The signatures of some interface methods have changed, for example:
  - Unified optional parameter naming. For example, previously all `Click` used `PageClickOptions`, now
    ```go
    Locator.Click(options ...LocatorClickOptions)
    Page.Click(selector string, options ...PageClickOptions)
    Frame.Click(selector string, options ...FrameClickOptions)
    ```
  - Some parameters are generated as new types or names, e.g. `Page.AddInitScript(script Script)`
- Removed several methods that were not documented upstream. These methods can be safely converted to other existing methods.  For example
  - `Page.ExpectedDialog` can use `Page.OnDialog` instead.
  - All `xxxAssertions.NotTo...`, use `.Not().xxx` pls.

### CODE REVIEW

File: examples/video/main.go [hunk 1]
```diff
@@ -19,9 +19,9 @@ func main() {
 	if err != nil {
 		log.Fatalf("could not launch Chromium: %v", err)
 	}
-	page, err := browser.NewPage(playwright.BrowserNewContextOptions{
+	page, err := browser.NewPage(playwright.BrowserNewPageOptions{
 		RecordVideo: &playwright.RecordVideo{
-			Dir: playwright.String("videos/"),
```
Comment (mxschmitt) [inline examples/video/main.go:24]: I in the past used it to differentiate if a user did set a string, since in Go a String can't be nil.

**Comment by mxschmitt** [comment]
Impressive work, thanks for putting effort into this.

File: examples/video/main.go (hunk 1 above)
Comment (canstand) [reply examples/video/main.go:24]: According to the documentation, recordVideo itself can be `nil`, and `path` is required once the recordVideo parameter is provided.  When generating type members, the script will judge whether the primitive type needs to be referenced by pointer or not.

```markdown
## context-option-recordvideo
* langs: js
- `recordVideo` <[Object]>
  - `dir` <[path]> Path to the directory to put videos into.
  - `size` ?<[Object]> Optional dimensions of the recorded videos. If not specified the size will be equal to `viewport`
    scaled down to fit into 800x800. If `viewport` is not configured explicitly the video size defaults to 800x450.
    Actual picture of each page will be scaled down if necessary to fit the specified size.
    - `width` <[int]> Video frame width.
    - `height` <[int]> Video frame height.
```

File: examples/screenshot/main.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func main() {
 	if err != nil {
 		log.Fatalf("could not create page: %v", err)
 	}
-	if _, err = page.Goto("http://whatsmyuseragent.org/", playwright.PageGotoOptions{
+	if _, err = page.Goto("https://playwright.dev/", playwright.PageGotoOptions{
 		// networkidle is DISCOURAGED
```
[the remark this answers is not in the document]
Comment (canstand) [reply examples/screenshot/main.go:26]: Will do this in next pr.

**Pull request merged**